packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -52,18 +51,19 @@
     ? flatten(fieldState?.value) ?? []
     : flatten(fieldState?.value)?.[0]
   $: component = multiselect ? CoreMultiselect : CoreSelect
-  $: expandedDefaultValue = expand(defaultValue)
   $: primaryDisplay = primaryDisplay || tableDefinition?.primaryDisplay
 
-  let optionsObj = {}
-  let initialValuesProcessed
+  let optionsObj
 
   $: {
-    if (!initialValuesProcessed && primaryDisplay) {
+    if (primaryDisplay && fieldState && !optionsObj) {
       // Persist the initial values as options, allowing them to be present in the dropdown,
       // even if they are not in the inital fetch results
-      initialValuesProcessed = true
-      optionsObj = (fieldState?.value || []).reduce((accumulator, value) => {
+      const valueAsSafeArray =
+        (!fieldState.value || Array.isArray(fieldState.value)
+          ? fieldState.value
+          : [fieldState.value]) || []
```

Comment:
I found this quite hard logic to follow. It looks like you're trying to ensure `fieldState.value` is an array without accidentally creating an array like `[undefined]`? Might be worth creating a function specifically for this, though I wouldn't block you on it.